packages/plugin-api-docgen/src/docgen.ts

```diff
@@ -81,7 +81,11 @@ export const docgen = async ({
             };
             let fileParser = withDefaultConfig(parserOpts);
             if (tsconfigPath) {
-              fileParser = withCustomConfig(tsconfigPath, parserOpts);
+              if (typeof tsconfigPath === 'string') {
+                fileParser = withCustomConfig(tsconfigPath, parserOpts);
+              } else if (typeof tsconfigPath === 'object') {
+                fileParser = withCustomConfig(tsconfigPath[key], parserOpts);
```

Comment:
We should also judge whether `tsconfigPath[key]` is existed or not here.

Add also for `compilerOptions` config below, same logic can be supported too.